src/cpp/src/tokenizer/tokenizer.cpp

```diff
@@ -934,11 +943,17 @@ Vocab Tokenizer::get_vocab() const {
 }
 
 const std::vector<std::string>& Tokenizer::get_vocab_vector() const {
+    OPENVINO_ASSERT(m_pimpl != nullptr, "Tokenizer is not initialized. Please check if the tokenizer model was provided and loaded correctly.");
     OPENVINO_ASSERT(!m_pimpl->m_vocab.empty(), "Tokenizer vocab is empty. Please check if the detokenizer model was provided and loaded correctly.");
     return m_pimpl->m_vocab;
 }
 
+std::shared_ptr<StructuredOutputController> Tokenizer::get_structured_output_controller() const {
+    return m_structured_output_controller;
+}
```

Comment:
I think we need to init the controller lazily, the first time `get_structured_output_controller` is called.